Octokit/Clients/TeamsClient.cs

```diff
@@ -37,6 +38,28 @@ public Task<Team> Get(int id)
             return ApiConnection.Get<Team>(endpoint, null, AcceptHeaders.NestedTeamsPreview);
         }
 
+        /// <summary>
+        /// Gets a single <see cref="Team"/> by name.
+        /// </summary>
+        /// <remarks>
+        /// this is more expensive than using the team ID, as first we
+        /// perform a lookup to determine matching teams.
+        /// </remarks>
+        /// <param name="orgName">The organization name.</param>
+        /// <param name="teamName">The team name.</param>
+        /// <exception cref="ArgumentNullException">if the list of teams returned from GetAll is null.</exception>
+        /// <exception cref="InvalidOperationException">if no matching team name is found.</exception>
+        /// <returns>The <see cref="Team"/> with the given identifier.</returns>
+        public async Task<Team> Get(string orgName, string teamName)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(orgName, nameof(orgName));
+            Ensure.ArgumentNotNullOrEmptyString(teamName, nameof(teamName));
+
+            var allTeams = await this.GetAll(orgName);
```

Comment:
By hiding `GetAll` here we don't let the caller control pagination, which means for orgs with a large # of teams it'll take several requests to fetch them all, before doing the filtering.